controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -489,6 +468,42 @@ func (r *StorageConsumerReconciler) reconcileConsumerConfigMap(
 				return err
 			}
 		}
+
+		if controllerRef.UID == r.storageConsumer.UID {
+			if consumerConfigMap.Data == nil {
+				consumerConfigMap.Data = map[string]string{}
+			}
+			defaultConsumerResourceNames := util.GetStorageConsumerDefaultResourceNames(
+				r.storageConsumer.Name,
+				string(r.storageConsumer.UID),
+				availableServices,
+			)
+			for key := range defaultConsumerResourceNames {
+				consumerConfigMap.Data[key] = cmp.Or(
+					strings.Trim(consumerConfigMap.Data[key], " "),
+					defaultConsumerResourceNames[key],
+				)
+			}
+		}
+
+		annotations := consumerConfigMap.GetAnnotations()
+		primaryConsumerUID, hasAnnotation := annotations[PrimaryConsumerUIDAnnotation]
+		if !hasAnnotation {
+			util.AddAnnotation(consumerConfigMap, PrimaryConsumerUIDAnnotation, string(r.storageConsumer.UID))
+		} else {
+			// Check if the consumer with the UID in the annotation exists; if not, update the annotation to the current consumer UID
+			consumerList := &ocsv1alpha1.StorageConsumerList{}
```

Comment:
You don't need to list, the idea here (as per Ohad) is, when an object is removed k8s will remove the owner and you just need to check in current owner does the annotation UID exists or not.